docs/guidelines.md

```diff
@@ -0,0 +1,43 @@
+# Plugin Author Guidelines
+
+Netlify plugins listed in the plugins.json file of this repository can be installed directly within the Netlify UI from the plugins directory. To help ensure a consistent user experience, we've prepared the following guidelines for authors submitting plugins to the directory.
+
+Before submitting a plugin for the plugins directory, please read and follow the guidelines below.
+
+## Provide a zero-config default.
+
+Plugins with required `inputs` cannot be installed via the Netlify UI. Wherever possible, a plugin should include default options that allow the plugin to run without configuration. In cases where a plugin requires a unique value (such as an API key for a third-party service), configure the plugin to accept this value from a [build environment variable](https://docs.netlify.com/configure-builds/environment-variables). For more complex or customized configuration, users can install the plugin via the `netlify.toml` configuration file.
+
+## Provide a README.
+
+Every plugin in the Netlify plugins directory includes a link to the plugin README. This file should include:
+- A description of what the plugin does and why that might be useful.
+- Any required environment variables.
+- Instructions for installing via `netlify.toml` configuration file, including any optional `inputs`.
```

Comment:
We discussed with @erquhart that maybe plugins `README` should redirect to our documentation instead for plugin installation instruction?
For example, right now, we provide two installation ways: either UI, or `package.json` + `netlify.toml`. Installing with `netlify.toml` but without adding to `package.json` is documented in most plugins `README` though. They probably should be updated, and linking to our official documentation would help with future updates?

Reply:
I left the link out for this iteration because the link is about to change. I suppose I could still tell people to link to the installation docs without actually providing the link yet. 🤔